nautobot/core/tests/test_views.py

```diff
@@ -278,6 +297,31 @@ def test_sso_login_button_visible(self):
         sso_login_search_result = self.make_request()
         self.assertIsNotNone(sso_login_search_result)
 
+    @override_settings(HIDE_RESTRICTED_UI=True)
+    def test_routes_redirect_back_to_login_if_hide_restricted_ui_true(self):
+        """Assert that api docs and graphql redirects to login page if user is unauthenticated and HIDE_RESTRICTED_UI=True."""
+        self.client.logout()
+        url = "/api/docs/"
+        response = self.client.get(url)
+        self.assertHttpStatus(response, 302)
+        self.assertEqual(response.url, f"/login/?next={url}")
+        response_content = response.content.decode(response.charset).replace("\n", "")
+        # Assert `self.footer_links` not found in `response_content`
+        # Since self.assertNotInHTML is not available, Hence the workaround
+        with self.assertRaises(AssertionError):
+            self.assertInHTML(self.footer_links, response_content)
```

Comment:
Not sure this is testing what you think here - since it's a 302 response, I don't think it has any content to check. Would need to do `follow=true` on the `get()` call, I think?